src/EA.Iws.Domain/Movement/IMovementRepository.cs

```diff
@@ -20,5 +20,7 @@ public interface IMovementRepository
         Task<IEnumerable<Movement>> GetActiveMovements(Guid notificationId);
 
         void Add(Movement movement);
+
+        Task<int> GetHighestMovementNumber(Guid notificationId);
```

Comment:
I think instead of getting the highest movement number it should be getting the last one used. In most cases this will be the highest number, but they could be entered out of order by an internal user.